### DISCUSSION TITLE
I can't run "pyccel omp_test.py --openmp"

### DISCUSSION THREAD

**Discussion by madou-sow** [pyccel/pyccel#1040] [Q&A]
1- When i do  "pyccel omp_test.py --openmp"
2- I have this result
ERROR at shared library generation stage
Traceback (most recent call last):
  File "/usr/local/anaconda3/bin/pyccel", line 33, in <module>
    sys.exit(load_entry_point('pyccel==1.3.1', 'console_scripts', 'pyccel')())
  File "/home/mamadou/.local/lib/python3.8/site-packages/pyccel/commands/console.py", line 216, in pyccel
    execute_pyccel(filename,
  File "/home/mamadou/.local/lib/python3.8/site-packages/pyccel/codegen/pipeline.py", line 362, in execute_pyccel
    generated_filepath = create_shared_library(codegen,
  File "/home/mamadou/.local/lib/python3.8/site-packages/pyccel/codegen/python_wrapper.py", line 133, in create_shared_library
    sharedlib_filepath = src_compiler.compile_shared_library(wrapper_compile_obj,
  File "/home/mamadou/.local/lib/python3.8/site-packages/pyccel/codegen/compiling/compilers.py", line 393, in compile_shared_library
    self.run_command(cmd, verbose)
  File "/home/mamadou/.local/lib/python3.8/site-packages/pyccel/codegen/compiling/compilers.py", line 424, in run_command
    raise RuntimeError(err_msg)
RuntimeError: Failed to build module
/usr/local/anaconda3/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/bin/ld: cannot find -lgomp
/usr/local/anaconda3/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/bin/ld: cannot find -lgomp
collect2: error: ld returned 1 exit status

3 - In Jupyter-Notebook , i type "
from pyccel.stdlib.internal.openmp import omp_set_num_threads"
when i execute i have this result

 ModuleNotFoundError: No module named 'pyccel.stdlib.internal.openmp' 
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
/tmp/ipykernel_6256/1935389673.py in <module>
----> 1 from pyccel.stdlib.internal.openmp import omp_set_num_threads

ModuleNotFoundError: No module named 'pyccel.stdlib.internal.openmp'

4- I need help for installing pyccel in my laptop

**Comment by EmilyBourne** [comment]
Hello, thank you for your interest in pyccel.

If the command `pyccel omp_test.py --openmp` is raising errors from the pyccel codebase then it would seem that you have successfully installed pyccel. However the error `ModuleNotFoundError: No module named 'pyccel.stdlib.internal.openmp'` from the Jupyter-Notebook would seem to show that it has not been made available to the notebook. I am not sure what would cause this as I don't personally use Jupyter. Presumably you have restarted your notebook since installing pyccel?

Regarding the error when running `pyccel omp_test.py --openmp`, the error that you are seeing:
```
/usr/local/anaconda3/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/bin/ld: cannot find -lgomp
```
is a compiler error. It says that gcc cannot find the openmp library on your computer. You should check that you have open mp installed. It seems this dependence is missing from our README :(
What platform are you using? I may be able to find the installation command in our tests.

If you run pyccel in verbose mode then you will be able to see the compiler command that is being run.
The command is:
```
pyccel omp_test.py --openmp --verbose
```

You should get output resembling:
```
>>> treating :: pyccel.stdlib.internal.openmp
>>> treating :: /home/emily/Code/pyccel/pyccel/stdlib/internal/openmp.pyh
gfortran -O3 -funroll-loops -fPIC -fopenmp -c -I /home/emily/Code/test/tmp/__pyccel__ /home/emily/Code/test/tmp/__pyccel__/omp_test.f90 -o /home/emily/Code/test/tmp/__pyccel__/omp_test.o -J /home/emily/Code/test/tmp/__pyccel__
gfortran -O3 -funroll-loops -fPIC -fopenmp -c -I /home/emily/Code/test/tmp/__pyccel__ /home/emily/Code/test/tmp/__pyccel__/bind_c_omp_test.f90 -o /home/emily/Code/test/tmp/__pyccel__/bind_c_omp_test.o -J /home/emily/Code/test/tmp/__pyccel__
gcc -O3 -funroll-loops -fPIC -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -pthread -c -I /home/emily/Code/test/tmp/__pyccel__/cwrapper -I /usr/include/python3.8 -I /home/emily/.local/lib/python3.8/site-packages/numpy/core/include /home/emily/Code/test/tmp/__pyccel__/cwrapper/cwrapper.c -o /home/emily/Code/test/tmp/__pyccel__/cwrapper/cwrapper.o
gcc -O3 -funroll-loops -fPIC -fopenmp -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -pthread -c -I /home/emily/Code/test/tmp/__pyccel__/cwrapper -I /home/emily/Code/test/tmp/__pyccel__ -I /usr/include/python3.8 -I /home/emily/.local/lib/python3.8/site-packages/numpy/core/include /home/emily/Code/test/tmp/__pyccel__/omp_test_wrapper.c -o /home/emily/Code/test/tmp/__pyccel__/omp_test_wrapper.o
gfortran -shared -O3 -funroll-loops -fPIC -fopenmp -I /home/emily/Code/test/tmp/__pyccel__/cwrapper -I /home/emily/Code/test/tmp/__pyccel__ -I /usr/include/python3.8 -I /home/emily/.local/lib/python3.8/site-packages/numpy/core/include -L /usr/lib -L /usr/lib/python3.8/config-3.8-x86_64-linux-gnu -Wl,-rpath /usr/lib -Wl,-rpath /usr/lib/python3.8/config-3.8-x86_64-linux-gnu /home/emily/Code/test/tmp/__pyccel__/omp_test.o /home/emily/Code/test/tmp/__pyccel__/bind_c_omp_test.o /home/emily/Code/test/tmp/__pyccel__/cwrapper/cwrapper.o /home/emily/Code/test/tmp/__pyccel__/omp_test_wrapper.o -o /home/emily/Code/test/tmp/__pyccel__/omp_test.cpython-38-x86_64-linux-gnu.so -lm -lgomp -lpython3.8
> Shared library has been created: /home/emily/Code/test/tmp/omp_test.cpython-38-x86_64-linux-gnu.so
```
Most of these lines are compiler commands and the final command (the linking command) refers to the library `-lgomp`

With this output you should be able to test the compiler command without using pyccel

**Comment by madou-sow** [comment]
$ uname -ar
Linux port-lipn12 5.4.0-89-generic #100~18.04.1-Ubuntu SMP Wed Sep 29 10:59:42 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

# lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04.2 LTS (beaver-osp1-arya X39)
Release:	18.04
Codename:	bionic


$pip3 install --user mpi4py
Requirement already satisfied: mpi4py in /usr/local/anaconda3/lib/python3.8/site-packages (3.1.1)

# conda install -c anaconda intel-openmp
Collecting package metadata (current_repodata.json): done
Solving environment: done

# All requested packages already installed.

# conda list | grep ^mp
mpc                       1.1.0                h10f8cd9_1  
mpfr                      4.0.2                hb69a4c5_1  
mpi                       1.0                       mpich    conda-forge
mpi4py                    3.1.1            py38he865349_0    conda-forge
mpich                     3.4.2                external_0    conda-forge
mpl-scatter-density       0.7                        py_0  
mpmath                    1.2.1            py38h06a4308_0  


$ pyccel omp_test.py --openmp --verbose
```
>>> treating :: pyccel.stdlib.internal.openmp
>>> treating :: /home/../.local/lib/python3.8/site-packages/pyccel/stdlib/internal/openmp.pyh
gfortran -O3 -funroll-loops -fPIC -fopenmp -c -I /home/../test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__ /home/../test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__/omp_test.f90 -o /home/../test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__/omp_test.o -J /home/../test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__
gfortran -O3 -funroll-loops -fPIC -fopenmp -c -I /home/../test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__ /home/../test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__/bind_c_omp_test.f90 -o /home/../test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__/bind_c_omp_test.o -J /home/../test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__
gcc -O3 -funroll-loops -fPIC -fopenmp -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -pthread -B /usr/local/anaconda3/compiler_compat -Wl,--sysroot=/ -c -I /home/../test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__ -I /home/../test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__/cwrapper -I /usr/local/anaconda3/include/python3.8 -I /usr/local/anaconda3/lib/python3.8/site-packages/numpy/core/include /home/../test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__/omp_test_wrapper.c -o /home/../test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__/omp_test_wrapper.o
gfortran -shared -O3 -funroll-loops -fPIC -fopenmp -I /home/../test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__ -I /home/../test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__/cwrapper -I /usr/local/anaconda3/include/python3.8 -I /usr/local/anaconda3/lib/python3.8/site-packages/numpy/core/include -L /usr/local/anaconda3/lib -L /usr/local/anaconda3/lib/python3.8/config-3.8-x86_64-linux-gnu -Wl,-rpath /usr/local/anaconda3/lib -Wl,-rpath /usr/local/anaconda3/lib/python3.8/config-3.8-x86_64-linux-gnu /home/../test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__/omp_test.o /home/../test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__/cwrapper/cwrapper.o /home/../test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__/bind_c_omp_test.o /home/../test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__/omp_test_wrapper.o -o /home/../test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__/omp_test.cpython-38-x86_64-linux-gnu.so -lm -lgomp -lpython3.8

ERROR at shared library generation stage
Traceback (most recent call last):
  File "/usr/local/anaconda3/bin/pyccel", line 33, in <module>
    sys.exit(load_entry_point('pyccel==1.3.1', 'console_scripts', 'pyccel')())
  File "/home/../.local/lib/python3.8/site-packages/pyccel/commands/console.py", line 216, in pyccel
    execute_pyccel(filename,
  File "/home/../.local/lib/python3.8/site-packages/pyccel/codegen/pipeline.py", line 362, in execute_pyccel
    generated_filepath = create_shared_library(codegen,
  File "/home/../.local/lib/python3.8/site-packages/pyccel/codegen/python_wrapper.py", line 133, in create_shared_library
    sharedlib_filepath = src_compiler.compile_shared_library(wrapper_compile_obj,
  File "/home/../.local/lib/python3.8/site-packages/pyccel/codegen/compiling/compilers.py", line 393, in compile_shared_library
    self.run_command(cmd, verbose)
  File "/home/../.local/lib/python3.8/site-packages/pyccel/codegen/compiling/compilers.py", line 424, in run_command
    raise RuntimeError(err_msg)
RuntimeError: Failed to build module
/usr/local/anaconda3/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/bin/ld: cannot find -lgomp
/usr/local/anaconda3/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/bin/ld: cannot find -lgomp
collect2: error: ld returned 1 exit status
```

$ anaconda -V
anaconda Command line client (version 1.8.0)

$ conda info 

     active environment : None
            shell level : 0
       user config file : /root/.condarc
 populated config files : /root/.condarc
          conda version : 4.10.3
    conda-build version : 3.21.5
         python version : 3.8.11.final.0
       virtual packages : __linux=5.4.0=0
                          __glibc=2.27=0
                          __unix=0=0
                          __archspec=1=x86_64
       base environment : /usr/local/anaconda3  (writable)
      conda av data dir : /usr/local/anaconda3/etc/conda
  conda av metadata url : None
           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /usr/local/anaconda3/pkgs
                          /root/.conda/pkgs
       envs directories : /usr/local/anaconda3/envs
                          /root/.conda/envs
               platform : linux-64
             user-agent : conda/4.10.3 requests/2.26.0 CPython/3.8.11 Linux/5.4.0-89-generic ubuntu/18.04.6 glibc/2.27
                UID:GID : 0:0
             netrc file : None

**Comment by EmilyBourne** [comment]
> conda install -c anaconda intel-openmp

If you have installed intel's openmp then I think you will need to request the intel compiler.
This can be done with:
```
pyccel omp_test.py --openmp --compiler=intel
```

If you are using the gcc compiler then OpenMP can be installed on linux using the command:
```
sudo apt install libomp-dev libomp5
```
(see updated README: https://github.com/pyccel/pyccel#linux-debian-ubuntu-mint)

**Comment by EmilyBourne** [comment]
Note this problem is related to OpenMP not MPI

**Comment by EmilyBourne** [comment]
The command that is failing is:
```
gfortran -shared -O3 -funroll-loops -fPIC -fopenmp -I /home/../test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__ -I /home/../test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__/cwrapper -I /usr/local/anaconda3/include/python3.8 -I /usr/local/anaconda3/lib/python3.8/site-packages/numpy/core/include -L /usr/local/anaconda3/lib -L /usr/local/anaconda3/lib/python3.8/config-3.8-x86_64-linux-gnu -Wl,-rpath /usr/local/anaconda3/lib -Wl,-rpath /usr/local/anaconda3/lib/python3.8/config-3.8-x86_64-linux-gnu /home/../test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__/omp_test.o /home/../test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__/cwrapper/cwrapper.o /home/../test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__/bind_c_omp_test.o /home/../test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__/omp_test_wrapper.o -o /home/../test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__/omp_test.cpython-38-x86_64-linux-gnu.so -lm -lgomp -lpython3.8
```
If you have not yet deleted the folder `/home/../test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__/` then you can run this command without using python or pyccel

**Comment by madou-sow** [comment]
1 ====
pyccel omp_test.py --openmp --compiler=intel

ERROR at Fortran compilation stage
pyccel:
 |fatal [codegen]: omp_test.py| Could not find compiler (ifort)

2 ==== 
dpkg -l | grep ^ii | grep libomp

ii  libomp-dev                 5.0.1-1                                         amd64        LLVM OpenMP runtime - dev package
ii  libomp5:amd64          5.0.1-1                                         amd64        LLVM OpenMP runtime
ii  libompl-dev:amd64    1.2.1+ds1-1build1         amd64        Open Motion Planning Library (OMPL) development files
ii  libompl12:amd64             1.2.1+ds1-1build1                               amd64        Open Motion Planning Library (OMPL)


3 ===

titi@port-lipn12:~/cerin27092021/test-pyccel/usage-openmp-pyccel-emily/openmpi0$ pyccel omp_test.py --openmp

ERROR at shared library generation stage
Traceback (most recent call last):
  File "/usr/local/anaconda3/bin/pyccel", line 33, in <module>
    sys.exit(load_entry_point('pyccel==1.3.1', 'console_scripts', 'pyccel')())
  File "/home/titi/.local/lib/python3.8/site-packages/pyccel/commands/console.py", line 216, in pyccel
    execute_pyccel(filename,
  File "/home/titi/.local/lib/python3.8/site-packages/pyccel/codegen/pipeline.py", line 362, in execute_pyccel
    generated_filepath = create_shared_library(codegen,
  File "/home/titi/.local/lib/python3.8/site-packages/pyccel/codegen/python_wrapper.py", line 133, in create_shared_library
    sharedlib_filepath = src_compiler.compile_shared_library(wrapper_compile_obj,
  File "/home/titi/.local/lib/python3.8/site-packages/pyccel/codegen/compiling/compilers.py", line 393, in compile_shared_library
    self.run_command(cmd, verbose)
  File "/home/titi/.local/lib/python3.8/site-packages/pyccel/codegen/compiling/compilers.py", line 424, in run_command
    raise RuntimeError(err_msg)
RuntimeError: Failed to build module
/usr/local/anaconda3/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/bin/ld: cannot find -lgomp
/usr/local/anaconda3/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/bin/ld: cannot find -lgomp
collect2: error: ld returned 1 exit status

titi@port-lipn12:~/cerin27092021/test-pyccel/usage-openmp-pyccel-emily/openmpi0$ pyccel omp_test.py --openmp --compiler=intel

ERROR at Fortran compilation stage
pyccel:
 |fatal [codegen]: omp_test.py| Could not find compiler (ifort)

titi@port-lipn12:~/cerin27092021/test-pyccel/usage-openmp-pyccel-emily/openmpi0$ pyccel omp_test.py --openmp --verbose
>>> treating :: pyccel.stdlib.internal.openmp
>>> treating :: /home/titi/.local/lib/python3.8/site-packages/pyccel/stdlib/internal/openmp.pyh
gfortran -O3 -funroll-loops -fPIC -fopenmp -c -I /home/toto/test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__ /home/toto/test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__/omp_test.f90 -o /home/toto/test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__/omp_test.o -J /home/toto/test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__
gfortran -O3 -funroll-loops -fPIC -fopenmp -c -I /home/toto/test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__ /home/toto/test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__/bind_c_omp_test.f90 -o /home/toto/test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__/bind_c_omp_test.o -J /home/toto/test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__
gcc -O3 -funroll-loops -fPIC -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -pthread -B /usr/local/anaconda3/compiler_compat -Wl,--sysroot=/ -fopenmp -c -I /home/toto/test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__/cwrapper -I /home/toto/test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__ -I /usr/local/anaconda3/include/python3.8 -I /usr/local/anaconda3/lib/python3.8/site-packages/numpy/core/include /home/toto/test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__/omp_test_wrapper.c -o /home/toto/test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__/omp_test_wrapper.o
gfortran -shared -O3 -funroll-loops -fPIC -fopenmp -I /home/toto/test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__/cwrapper -I /home/toto/test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__ -I /usr/local/anaconda3/include/python3.8 -I /usr/local/anaconda3/lib/python3.8/site-packages/numpy/core/include -L /usr/local/anaconda3/lib -L /usr/local/anaconda3/lib/python3.8/config-3.8-x86_64-linux-gnu -Wl,-rpath /usr/local/anaconda3/lib -Wl,-rpath /usr/local/anaconda3/lib/python3.8/config-3.8-x86_64-linux-gnu /home/toto/test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__/bind_c_omp_test.o /home/toto/test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__/omp_test.o /home/toto/test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__/cwrapper/cwrapper.o /home/toto/test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__/omp_test_wrapper.o -o /home/toto/test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__/omp_test.cpython-38-x86_64-linux-gnu.so -lm -lpython3.8 -lgomp

ERROR at shared library generation stage
Traceback (most recent call last):
  File "/usr/local/anaconda3/bin/pyccel", line 33, in <module>
    sys.exit(load_entry_point('pyccel==1.3.1', 'console_scripts', 'pyccel')())
  File "/home/titi/.local/lib/python3.8/site-packages/pyccel/commands/console.py", line 216, in pyccel
    execute_pyccel(filename,
  File "/home/titi/.local/lib/python3.8/site-packages/pyccel/codegen/pipeline.py", line 362, in execute_pyccel
    generated_filepath = create_shared_library(codegen,
  File "/home/titi/.local/lib/python3.8/site-packages/pyccel/codegen/python_wrapper.py", line 133, in create_shared_library
    sharedlib_filepath = src_compiler.compile_shared_library(wrapper_compile_obj,
  File "/home/titi/.local/lib/python3.8/site-packages/pyccel/codegen/compiling/compilers.py", line 393, in compile_shared_library
    self.run_command(cmd, verbose)
  File "/home/titi/.local/lib/python3.8/site-packages/pyccel/codegen/compiling/compilers.py", line 424, in run_command
    raise RuntimeError(err_msg)
RuntimeError: Failed to build module
/usr/local/anaconda3/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/bin/ld: cannot find -lgomp
/usr/local/anaconda3/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/bin/ld: cannot find -lgomp
collect2: error: ld returned 1 exit status

**Comment by madou-sow** [comment]
tree
.
├── omp_test.cpython-36m-x86_64-linux-gnu.so
├── omp_test.py
├── prog_omp_test
└── __pyccel__
    ├── bind_c_omp_test.f90
    ├── bind_c_omp_test.mod
    ├── bind_c_omp_test.o
    ├── bind_c_omp_test.o.lock
    ├── cwrapper
    │   ├── cwrapper.c
    │   ├── cwrapper.h
    │   ├── cwrapper_ndarrays.c
    │   ├── cwrapper_ndarrays.h
    │   ├── cwrapper.o
    │   ├── cwrapper.o.lock
    │   └── numpy_version.h
    ├── cwrapper.lock
    ├── omp_test.f90
    ├── omp_test.mod
    ├── omp_test.o
    ├── omp_test.o.lock
    ├── omp_test_wrapper.c
    ├── omp_test_wrapper.o
    ├── omp_test_wrapper.o.lock
    ├── prog_omp_test.f90
    ├── prog_omp_test.lock
    └── prog_omp_test.o

2 directories, 25 files



./omp_test.cpython-36m-x86_64-linux-gnu.so -lm -lgomp -lpython3.8
Erreur de segmentation (core dumped)

**Comment by madou-sow** [comment]
> > conda install -c anaconda intel-openmp
> 
> If you have installed intel's openmp then I think you will need to request the intel compiler. This can be done with:
> 
> ```
> pyccel omp_test.py --openmp --compiler=intel
> ```
> 
> If you are using the gcc compiler then OpenMP can be installed on linux using the command:
> 
> ```
> sudo apt install libomp-dev libomp5
> ```
> 
> (see updated README: https://github.com/pyccel/pyccel#linux-debian-ubuntu-mint)

apt install libblas-dev liblapack-dev gfortran gcc libomp-dev libomp5 libopenmpi-dev openmpi-bin
Lecture des listes de paquets... Fait
Construction de l'arbre des dépendances       
Lecture des informations d'état... Fait
libblas-dev est déjà la version la plus récente (3.7.1-4ubuntu1).
liblapack-dev est déjà la version la plus récente (3.7.1-4ubuntu1).
libomp-dev est déjà la version la plus récente (5.0.1-1).
libomp5 est déjà la version la plus récente (5.0.1-1).
libopenmpi-dev est déjà la version la plus récente (2.1.1-8).
openmpi-bin est déjà la version la plus récente (2.1.1-8).
gcc est déjà la version la plus récente (4:7.4.0-1ubuntu2.3).
gfortran est déjà la version la plus récente (4:7.4.0-1ubuntu2.3).
0 mis à jour, 0 nouvellement installés, 0 à enlever et 8 non mis à jour.

**Comment by madou-sow** [comment]
> The command that is failing is:
> 
> ```
> gfortran -shared -O3 -funroll-loops -fPIC -fopenmp -I /home/../test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__ -I /home/../test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__/cwrapper -I /usr/local/anaconda3/include/python3.8 -I /usr/local/anaconda3/lib/python3.8/site-packages/numpy/core/include -L /usr/local/anaconda3/lib -L /usr/local/anaconda3/lib/python3.8/config-3.8-x86_64-linux-gnu -Wl,-rpath /usr/local/anaconda3/lib -Wl,-rpath /usr/local/anaconda3/lib/python3.8/config-3.8-x86_64-linux-gnu /home/../test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__/omp_test.o /home/../test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__/cwrapper/cwrapper.o /home/../test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__/bind_c_omp_test.o /home/../test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__/omp_test_wrapper.o -o /home/../test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__/omp_test.cpython-38-x86_64-linux-gnu.so -lm -lgomp -lpython3.8
> ```
> 
> If you have not yet deleted the folder `/home/../test-pyccel/usage-openmp-pyccel-emily/openmpi0/__pyccel__/` then you can run this command without using python or pyccel


tree
.
├── omp_test.cpython-36m-x86_64-linux-gnu.so
├── omp_test.py
├── prog_omp_test
└── pyccel
├── bind_c_omp_test.f90
├── bind_c_omp_test.mod
├── bind_c_omp_test.o
├── bind_c_omp_test.o.lock
├── cwrapper
│   ├── cwrapper.c
│   ├── cwrapper.h
│   ├── cwrapper_ndarrays.c
│   ├── cwrapper_ndarrays.h
│   ├── cwrapper.o
│   ├── cwrapper.o.lock
│   └── numpy_version.h
├── cwrapper.lock
├── omp_test.f90
├── omp_test.mod
├── omp_test.o
├── omp_test.o.lock
├── omp_test_wrapper.c
├── omp_test_wrapper.o
├── omp_test_wrapper.o.lock
├── prog_omp_test.f90
├── prog_omp_test.lock
└── prog_omp_test.o

2 directories, 25 files

./omp_test.cpython-36m-x86_64-linux-gnu.so -lm -lgomp -lpython3.8
Erreur de segmentation (core dumped)

**Comment by EmilyBourne** [comment]
1.
> Could not find compiler (ifort)

If you don't have an intel compiler installed then you cannot use intel's openmp:
> conda install -c anaconda intel-openmp

2. Great it looks like you have managed to install openmp :+1:

3/4.

I am not sure why you are getting `ERROR at shared library generation stage` messages, but you have managed to get a .so file. The error should mean that you weren't able to generate it so something you tried must have worked.

The .so file is not an executable. It is a python module. You can use it from python with:
```
import omp_test
```
If you want an executable then you need to add a main with:
```
if __name__ == '__main__':
```
This will then generate an executable `omp_test` which you can execute with `./omp_test` (you don't need to pass libraries when executing, only when compiling)

**Comment by EmilyBourne** [comment]
Hi, I'm just following up on this. Have you had any success?
